zjit/src/hir.rs

```diff
@@ -3325,13 +3333,17 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
                     let id = ID(get_arg(pc, 0).as_u64());
                     // ic is in arg 1
                     let exit_id = fun.push_insn(block, Insn::Snapshot { state: exit_state });
+                    // TODO: We only really need this if self_val is a class/module
```

Comment:
From just looking at the HIR code, it wasn't obvious to me why we need the patch point.

```suggestion
                    // Assume the single-Ractor mode to omit `gen_prepare_non_leaf_call` on `gen_getivar`
                    // TODO: We only really need this if self_val is a class/module
```